### ISSUE TITLE
HSCF.Communication.ScsServices.Communication.Messages.ScsRem oteException: A status graphic with that value set already exists Service Version: NO_VERSION

### ISSUE THREAD

**Issue by alexbk66** [HomeSeer/Plugin-SDK#169]
HSCF.Communication.ScsServices.Communication.Messages.ScsRem oteException: A status graphic with that value set already exists Service Version: NO_VERSION
---> System.ArgumentException: A status graphic with that value set already exists
at: HomeSeer.PluginSdk.Devices.StatusGraphicCollection.AddRange( List1 statusGraphics)
at: HomeSeer.PluginSdk.Devices.HsFeature.AddStatusGraphics(List 1 statusGraphics)
at: Scheduler.Util.Devices.Converters.ToModernFeature(DeviceClas s ind)
at: Scheduler.hsapplication.GetAllDevices(Boolean withFeatures)
--- End of inner exception stack trace ---
Server stack trace: Exception rethrown
at: [0]:
at: System.Runtime.Remoting.Proxies.RealProxy.HandleReturnMessag e(IMessage reqMsg, IMessage retMsg)
at: System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke(Mess ageData& msgData, Int32 type)
at: HomeSeer.PluginSdk.IHsController.GetAllDevices(Boolean withFeatures)
at: HSPI_AKSmartDevice.Controller.Update()
at: PluginLib.ControllerBase.Update(Boolean InitIOinThread)

**Comment by spudwebb** [comment]
tracked as HS-1129

**Comment by alexbk66** [comment]
Also, minor - is it possible to remove the `Service Version: NO_VERSION`

**Comment by alexbk66** [comment]
Same for `GetDeviceWithFeaturesByRef`...  Just wondering - do you actually do any testing???

```
2/19/2021 5:55:41 PM	AK SmartDevice	Error	GetDeviceWithFeaturesByRef(69): Exception while executing GetDeviceWithFeaturesByRef - devRef : 69 - message : Error while converting 70 to HsFeature : A status graphic with that value set already exists
Service Version: NO_VERSION
2/19/2021 5:55:41 PM	AK SmartDevice	Error	HSCF.Communication.ScsServices.Communication.Messages.ScsRemoteException: Exception while executing GetDeviceWithFeaturesByRef - devRef : 69 - message : Error while converting 70 to HsFeature : A status graphic with that value set already exists
Service Version: NO_VERSION ---> System.Exception: Exception while executing GetDeviceWithFeaturesByRef - devRef : 69 - message : Error while converting 70 to HsFeature : A status graphic with that value set already exists ---> System.Exception: Error while converting 70 to HsFeature : A status graphic with that value set already exists ---> System.ArgumentException: A status graphic with that value set already exists
   at: HomeSeer.PluginSdk.Devices.StatusGraphicCollection.AddRange(List`1 statusGraphics)
   at: HomeSeer.PluginSdk.Devices.HsFeature.AddStatusGraphics(List`1 statusGraphics)
   at: Scheduler.Util.Devices.Converters.ToModernFeature(DeviceClass ind)
   at: Scheduler.hsapplication.GetDeviceWithFeaturesByRef(Int32 devRef)
   --- End of inner exception stack trace ---
   at: Scheduler.hsapplication.GetDeviceWithFeaturesByRef(Int32 devRef)
   --- End of inner exception stack trace ---
   at: Scheduler.hsapplication.GetDeviceWithFeaturesByRef(Int32 devRef)
   --- End of inner exception stack trace ---

Server stack trace: 

Exception rethrown at: [0]: 
   at: System.Runtime.Remoting.Proxies.RealProxy.HandleReturnMessage(IMessage reqMsg, IMessage retMsg)
   at: System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke(MessageData& msgData, Int32 type)
   at: HomeSeer.PluginSdk.IHsController.GetDeviceWithFeaturesByRef(Int32 devRef)
   at: HSPI_AKSmartDevice.Controller.Update()
```

**Comment by tenholde** [comment]
This occurs when you have a device defined with overlapping (invalid) graphic pairs. Usually from an hs3 plug-in.  Still a bug, but I can see this not being caught in testing. Until bug is fixed, fix the bad device

**Comment by alexbk66** [comment]
It's not me - my user reported.  And there's too many devices to fix, see attached log
[Uploading log.txt…]()

**Comment by spudwebb** [comment]
This is fixed in HS 4.1.14.0  GetAllDevices will no longer throw an exception if one device is badly configured, instead it will ignore it and log a warning.

Also, "Service Version: NO_VERSION" has been removed from the exception message
